engines/bops_admin/app/controllers/bops_admin/tokens_controller.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module BopsAdmin
+  class TokensController < ApplicationController
+    before_action :set_tokens, only: %i[index]
+    before_action :build_token, only: %i[new create]
+    before_action :set_token, only: %i[edit update destroy]
+
+    def index
+      respond_to do |format|
+        format.html
+      end
+    end
+
+    def new
+      respond_to do |format|
+        format.html
+      end
+    end
+
+    def edit
+      respond_to do |format|
+        format.html
+      end
+    end
+
+    def create
+      respond_to do |format|
+        if @token.save
+          format.html { render :show }
```

Comment:
Normally you'd redirect here but we render so we can show the token a single time